crates/apub/src/protocol/objects/mod.rs

```diff
@@ -99,6 +106,57 @@ impl LanguageTag {
   }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub(crate) struct PersonOrGroupModerators(Url);
+
+impl Deref for PersonOrGroupModerators {
+  type Target = Url;
+
+  fn deref(&self) -> &Self::Target {
+    &self.0
+  }
+}
+
+impl From<DbUrl> for PersonOrGroupModerators {
+  fn from(value: DbUrl) -> Self {
+    PersonOrGroupModerators(value.into())
+  }
+}
+
+impl PersonOrGroupModerators {
+  pub(crate) fn creator(&self) -> ObjectId<ApubPerson> {
+    self.deref().clone().into()
+  }
+
+  pub(crate) fn moderators(&self) -> CollectionId<ApubCommunityModerators> {
+    self.deref().clone().into()
+  }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(untagged)]
+pub(crate) enum AttributedTo {
+  Lemmy(PersonOrGroupModerators),
+  Peertube(Vec<AttributedToPeertube>),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct AttributedToPeertube {
+  #[serde(rename = "type")]
+  pub kind: PersonOrGroupType,
+  pub id: ObjectId<UserOrCommunity>,
+}
+
+impl AttributedTo {
+  pub(crate) fn url(self) -> Option<DbUrl> {
+    match self {
+      AttributedTo::Lemmy(l) => Some(l.moderators().into()),
+      AttributedTo::Peertube(_) => None,
+    }
```

Comment:
Nitpick: could be:
```suggestion
    if let AttributedTo::Lemmy(l) = self {
        Some(l.moderators().into())
    } else {
        None
    }
```

I'll leave it to the other reviewers to decide which style is preferred.

Reply:
Match has less lines which makes it easier to read, plus we might add more variants in the future. So leave it as is.